task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,281 @@
+
+            # Translate the output format
+            TRANSLATE_EXIT_CODE=0
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|' || TRANSLATE_EXIT_CODE=$?
+            if [[ "${TRANSLATE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            else
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+                --strip-path-prefix="${SOURCE_CODE_DIR}"/source/
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as sast-unicode-check only provides an URI to embed the context
+            csgrep "${CSGERP_OPTS[@]}" raw_sast_unicode_check_out.txt > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        csfilter_kfp()
+        {
+            echo "Message: Filtering false positives in results files using ${KFP_GIT_URL}..." >&2
+
+            # Build initial csfilter-kfp command
+            csfilter_kfp_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
+
+            # Append --project-nvr option if PROJECT_NVR is set
+            if [[ -n "${PROJECT_NVR}" ]]; then
+                csfilter_kfp_cmd+=" --project-nvr="${PROJECT_NVR}""
+            fi
+
+            # Append --record-excluded option if RECORD_EXCLUDED is true
+            if [[ "${RECORD_EXCLUDED}" == "true" ]]; then
+                csfilter_kfp_cmd+=" --record-excluded=raw-excluded-findings.json"
+            fi
+
+            KFP_EXIT_CODE=0
+            ${csfilter_kfp_cmd[@]} processed_sast_unicode_check_out.json > sast_unicode_check_out.json || KFP_EXIT_CODE=$?
+            if [ "${KFP_EXIT_CODE}" -eq 0 ]; then
+                echo "Message: Succeed to filter known false positives" >&2
+            else
+                echo "Error: failed to filter known false positives" >&2
+                return 1
+            fi
+
+            if [ -f "raw-excluded-findings.json" ]; then
+                csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ raw-excluded-findings.json > excluded-findings.json
+            fi
+        }
+
+        generate_sarif_report()
+        {
+            csgrep --mode=sarif sast_unicode_check_out.json > sast_unicode_check_out.sarif
+            if [[ "${FUC_EXIT_CODE}" -eq 0 ]]; then
+                note="Task $(context.task.name) success: No finding was detected"
+                ERROR_OUTPUT=$(make_result_json -r SUCCESS -t "$note")
```

Comment:
Has this been tested? It seems there is an error in the logs:
`[sast-unicode-check] + note='Unexpected error: Script errored at command: return 1.'`

Reply:
the latest pipeline result: https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/chhan/applications/unicode-control-test/pipelineruns/find-unicode-control-on-pull-request-l9xrn
but should add install pacakges and certificated steps
```
        # Only used for testing
        microdnf install dnf -y
        dnf install -y 'dnf-command(copr)'
        dnf copr enable copr.devel.redhat.com/@osh/csdiff -y
        dnf install -y csdiff git python3-file-magic

        # Installation of Red Hat certificates for cloning Red Hat internal repositories
        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
        update-ca-trust
```